packages/@ember/object/type-tests/computed.test.ts

```diff
@@ -0,0 +1,28 @@
+import { computed } from '@ember/object';
+
+class Foo {
+  declare firstName: string;
+  declare lastName: string;
+
+  @computed('firstName', 'lastName')
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+
+  @(computed('firstName', 'lastName').readOnly())
+  get readonlyFullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+
+  // NOTE: This should probably not work, but types do currently allow it.
```

Comment:
Iiiiiinteresting. I don't know that there's a way we could do otherwise, either. 🤔